easybuild/easyblocks/t/tensorflow.py

```diff
@@ -152,6 +152,87 @@ def write_wrapper(self, wrapper_dir, compiler, i_mpi_root):
+        """
+        def is_version_ok(dep_version):
+            version = LooseVersion(self.version)
+            min_version = LooseVersion(dep_version[0])
+            result = False
+            if version >= min_version:
+                # min. version matches, so OK if max version matches or not present
+                if len(dep_version) == 1 or version < LooseVersion(dep_version[1]):
+                    result = True
+            return result
+
+        def make_tf_name(eb_dep_name, tf_dep_name):
+            return tf_dep_name or eb_dep_name.lower().replace('-', '_')
+
+        # For this list check third_party/systemlibs/syslibs_configure.bzl --> VALID_LIBS
+        # Also verify third_party/systemlibs/<name>.BUILD if it does something "strange" (e.g. link hardcoded headers)
+        available_system_libs = (
+            # Format: 1. dependency name in EB
+            #         2. name in TF or None to lowercase EB name with - replaced by _
+            #         3. min. (incl), max. (excl) TF version required
+            ('cURL', None, ('2.1.0',)),
+            ('double-conversion', None, ('2.1.0',)),
+            ('giflib', 'gif', ('2.1.0',)),
+            ('hwloc', None, ('2.1.0',)),
+            ('ICU', None, ('2.1.0',)),
+            ('JsonCpp', 'jsoncpp_git', ('2.1.0',)),
+            ('libjpeg-turbo', 'jpeg', ('2.1.0', '2.2.0')),
+            ('libjpeg-turbo', 'libjpeg_turbo', ('2.2.0', )),
+            ('LMDB', None, ('2.1.0',)),
+            ('NASM', None, ('2.1.0',)),
+            ('SQLite', 'org_sqlite', ('2.1.0',)),
+            ('PCRE', None, ('2.1.0',)),
+            ('libpng', 'png', ('2.1.0',)),
+            ('pybind11', None, ('2.2.0',)),
+            ('snappy', None, ('2.1.0',)),
+            ('SWIG', None, ('2.1.0',)),
+            ('zlib', 'zlib_archive', ('2.1.0', '2.2.0')),
+            ('zlib', 'zlib', ('2.2.0',)),
+        )
+        # Same as above but installed as python extensions
+        python_system_libs = (
+            ('absl-py', None, ('2.1.0',)),
+            ('astor', 'astor_archive', ('2.1.0',)),
+            ('astunparse', 'astunparse_archive', ('2.2.0',)),
+            ('gast', 'gast_archive', ('2.1.0',)),
+            ('Keras-Applications', 'keras_applications_archive', ('2.1.0', '2.2.0')),
+            ('opt-einsum', 'opt_einsum_archive', ('2.1.0',)),
+            ('google-pasta', 'pasta', ('2.1.0',)),
+            ('termcolor', 'termcolor_archive', ('2.1.0',)),
+        )
+
+        dependencies = set(dep['name'] for dep in self.cfg.dependencies())
+
+        system_libs = []
+        cpaths = []
+        libpaths = []
+        for dep_name, tf_dep_name, version_requirement in available_system_libs:
+            if dep_name in dependencies and is_version_ok(version_requirement):
+                system_libs.append(make_tf_name(dep_name, tf_dep_name))
+                sw_root = get_software_root(dep_name)
```

Comment:
We should probably verify here if this is different from `None`, since due some dependencies may be being filtered (via `--filter-deps`). See also https://github.com/easybuilders/easybuild-framework/issues/1494